src/field.cpp

```diff
@@ -1661,6 +1669,20 @@ bool map::process_fields_in_submap( submap *const current_submap,
                         make_rubble( p );
                         break;
 
+                    case fd_fungicidal_gas:
+                        {
+                            dirty_transparency_cache = true;
+                            //spread the chemical agent as fast as smoke would
+                            spread_gas( cur, p, curtype, 80, 40 );
+                            //check the terrain and replace it accordingly to simulate the fungus dieing off
+                            const auto &ter = map_tile.get_ter_t();
+                            if( ter.has_flag( "FUNGUS" ) ) {
+                                ter_set( p, t_dirt );
+                                furn_set( p, f_null );
```

Comment:
You need to have separate check for furniture. Otherwise when a fungal terrain has a non-fungal furniture, it will be destroyed too.